oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProvider.java

```diff
@@ -183,7 +195,26 @@ private OidcClientRegistrationAuthenticationToken registerClient(OidcClientRegis
 		}
 
 		RegisteredClient registeredClient = this.registeredClientConverter.convert(clientRegistrationAuthentication.getClientRegistration());
-		this.registeredClientRepository.save(registeredClient);
+
+		// Copy RegisteredClient and encode only secret
+		String rawClientSecret = registeredClient.getClientSecret();
+		String clientSecret = null;
+		if (rawClientSecret != null) {
+			clientSecret = passwordEncoder.encode(rawClientSecret);
+		}
+		RegisteredClient saveRegisteredClient = RegisteredClient.withId(registeredClient.getId())
```

Comment:
Use `RegisteredClient.from(registeredClient)` to make a copy and then update `clientSecret`. Also, the `RegisteredClient` should only be copied/updated if the `clientSecret` is set, otherwise it should use the original returned from the converter.